internal/datasourcev2/streaming_data_source.go

```diff
@@ -438,81 +453,74 @@ func (sp *StreamProcessor) GetFilterKey() string {
 func deserializeEvents(events []es.Event) ([]fdv2proto.Event, error) {
 	updates := make([]fdv2proto.Event, 0, len(events))
 
-	parseItem := func(r jreader.Reader, kind datakinds.DataKindInternal) (ldstoretypes.ItemDescriptor, error) {
-		item, err := kind.DeserializeFromJSONReader(&r)
+	parseItem := func(r jreader.Reader, kind fdv2proto.ObjectKind) (ldstoretypes.ItemDescriptor, error) {
+		dataKind, err := kind.ToFDV1()
+		if err != nil {
+			return ldstoretypes.ItemDescriptor{}, err
+		}
+		item, err := dataKind.DeserializeFromJSONReader(&r)
 		return item, err
 	}
 
 	for _, event := range events {
 		switch fdv2proto.EventName(event.Event()) {
 		case fdv2proto.EventPutObject:
 			r := jreader.NewReader([]byte(event.Data()))
-			// var version int
-			var dataKind datakinds.DataKindInternal
-			var key string
-			var item ldstoretypes.ItemDescriptor
-			var err error
 
-			for obj := r.Object().WithRequiredProperties([]string{versionField, kindField, keyField, "object"}); obj.Next(); {
+			var (
+				kind    fdv2proto.ObjectKind
+				key     string
+				version int
+				item    ldstoretypes.ItemDescriptor
+				err     error
+			)
+
+			for obj := r.Object().WithRequiredProperties([]string{versionField, kindField, keyField, objectField}); obj.Next(); {
 				switch string(obj.Name()) {
 				case versionField:
-					// version = r.Int()
+					version = r.Int()
 				case kindField:
-					kind := r.String()
-					dataKind = dataKindFromKind(kind)
-					if dataKind == nil {
-						//nolint: godox
-						// TODO: We are skipping here without showing a warning. Need to address that later.
-						continue
-					}
+					kind = fdv2proto.ObjectKind(r.String())
+					//nolint:godox
+					// TODO: An unrecognized kind should be ignored for forwards compat; the question is,
```

Comment:
That's a good question. Generally I like to handle error cases as close to the source of the error as possible. In this case, I think I can see the argument for saying it isn't an error in the data source. It's only an error once you try to apply it to something that doesn't understand it. 

When we go to update our SDKs in the future to support a third kind, would I expect to look at the data source or the data store? For that reason alone, I think we can let the store handle this.

Reply:
That makes sense to me.